src/Cli/Microsoft.DotNet.Cli.Utils/MSBuildForwardingAppWithoutLogging.cs

```diff
@@ -205,6 +199,16 @@ private static string GetDotnetPath()
             return new Muxer().MuxerPath;
         }
 
+        internal static Dictionary<string, string> GetMSBuildRequiredEnvironmentVariables()
```

Comment:
This call is now being used in a few places, sometimes to create a new dictionary and sometimes to just iterate this data which is effectively static. Did you consider instead having a definition like this: 

```csharp
 internal static ImmutableDictionary<string, string> MSBuildRequiredEnvironmentVariables { get; } = ...
```

Then in the places where a new dictionary is needed use `new (MSBuildRequiredEnvironmentVariables)` and for iteration it can just use the property directly.

Reply:
The default env vars cannot be static and initialized only once, they can change.

Reply:
You can still make it static and mutable so that you only have to reinitialize it when it does change, right?